src/hooks/use-sync-states-progress-info.ts

```diff
@@ -130,12 +137,59 @@ export function useSyncStatesProgressInfo() {
 		[]
 	);
 
+	const getBackupStatusWithProgress = useCallback(
+		(
+			hasBackupCompleted: boolean,
+			pullStatesProgressInfo: PullStateProgressInfoValues,
+			response: SyncBackupResponse
+		) => {
+			const frontendStatus = hasBackupCompleted
+				? pullStatesProgressInfo.downloading.key
+				: response.status;
+			let newProgressInfo: PullStateProgressInfo | null = null;
+			if ( response.status === 'in-progress' ) {
+				newProgressInfo = pullStatesProgressInfo[ frontendStatus ];
+				newProgressInfo.progress =
+					IN_PROGRESS_INITIAL_VALUE + IN_PROGRESS_TO_DOWNLOADING_STEP * ( response.percent / 100 );
+			}
+			const statusWithProgress =
+				newProgressInfo ||
+				pullStatesProgressInfo[ frontendStatus ] ||
+				pullStatesProgressInfo.failed;
+
+			return statusWithProgress;
+		},
+		[]
+	);
+
+	const getPullStatusWithProgress = useCallback(
+		( sitePullState?: PullStateProgressInfo, importState?: ImportProgressState[ string ] ) => {
+			if ( ! importState && sitePullState ) {
+				return { message: sitePullState.message, progress: sitePullState.progress };
+			}
+			if ( importState ) {
+				if ( importState.progress === 100 ) {
+					return { message: __( 'Applying final details…' ), progress: 99 };
+				}
+				return {
+					message: importState.statusMessage,
+					progress:
+						IMPORTING_INITIAL_VALUE + IMPORTING_TO_FINISHED_STEP * ( importState.progress / 100 ),
+				};
+			}
```

Comment:
```suggestion
			if ( importState ) {
				if ( importState.progress === 100 ) {
					return { message: __( 'Applying final details…' ), progress: 99 };
				}
				return {
					message: importState.statusMessage,
					progress:
						IMPORTING_INITIAL_VALUE + IMPORTING_TO_FINISHED_STEP * ( importState.progress / 100 ),
				};
			}
			if ( sitePullState ) {
				return { message: sitePullState.message, progress: sitePullState.progress };
			}
```
Nitpicking, but changing the order of the `if` statements clarifies intent.

Reply:
I agree, the suggesions has less conditions